test/virtual/include/openenclave/edger8r/common.h

```diff
@@ -66,35 +66,46 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
     return result;
 }
 
-#define OE_ADD_SIZE(total, size)                                   \
-    do                                                             \
-    {                                                              \
-        if (sizeof(total) > sizeof(size_t) && total > OE_SIZE_MAX) \
-        {                                                          \
-            _result = OE_INVALID_PARAMETER;                        \
-            goto done;                                             \
-        }                                                          \
-        if (sizeof(size) > sizeof(size_t) && size > OE_SIZE_MAX)   \
-        {                                                          \
-            _result = OE_INVALID_PARAMETER;                        \
-            goto done;                                             \
-        }                                                          \
-        if (oe_add_size((size_t*)&total, (size_t)size) != OE_OK)   \
-        {                                                          \
-            _result = OE_INTEGER_OVERFLOW;                         \
-            goto done;                                             \
-        }                                                          \
+#define OE_MUL_SIZE(a, b, c)                         \
+    do                                               \
+    {                                                \
+        if ((a) && (b) && ((a) > OE_SIZE_MAX / (b))) \
+        {                                            \
+            _result = OE_INTEGER_OVERFLOW;           \
+            goto done;                               \
+        }                                            \
+        (c) = (a) * (b);                             \
+    } while (0)
```

Comment:
Consider if it is better to use [safemath.h](https://github.com/openenclave/openenclave/blob/5aea07478409fec7db0695180aa0ba3efbaf7a43/include/openenclave/internal/safemath.h#L91) or copy relevant portions of code from that header

Reply:
We enforce the `_a` and `_b` to be `size_t`. Therefore, we only need one check (out of four) from https://github.com/openenclave/openenclave/blob/92d4ad31004c24c104968906b7505aeeb6f20459/include/openenclave/internal/safemath.h#L94.